bigtable/hbase/snippets/src/test/java/com/example/bigtable/ReadsTest.java

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.bigtable;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertNotNull;
+
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminClient;
+import com.google.cloud.bigtable.admin.v2.models.CreateTableRequest;
+import com.google.cloud.bigtable.data.v2.BigtableDataClient;
+import com.google.cloud.bigtable.data.v2.models.BulkMutation;
+import com.google.cloud.bigtable.data.v2.models.Mutation;
+import com.google.protobuf.ByteString;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.UUID;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runners.MethodSorters;
+
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class ReadsTest {
+
+  private static final String INSTANCE_ENV = "BIGTABLE_TESTING_INSTANCE";
+  private static final String TABLE_ID =
+      "mobile-time-series-" + UUID.randomUUID().toString().substring(0, 20);
+  private static final String COLUMN_FAMILY_NAME = "stats_summary";
+  private static final long TIMESTAMP = System.currentTimeMillis();
+  private static final long TIMESTAMP_NANO = TIMESTAMP * 1000;
+
+  private static String projectId;
+  private static String instanceId;
+  private ByteArrayOutputStream bout;
+
+  private static String requireEnv(String varName) {
+    String value = System.getenv(varName);
+    assertNotNull(
+        String.format("Environment variable '%s' is required to perform these tests.", varName),
+        value);
+    return value;
+  }
+
+  @BeforeClass
+  public static void beforeClass() throws IOException {
+    projectId = requireEnv("GOOGLE_CLOUD_PROJECT");
+    instanceId = requireEnv(INSTANCE_ENV);
+
+    try (BigtableTableAdminClient adminClient =
+        BigtableTableAdminClient.create(projectId, instanceId)) {
+      CreateTableRequest createTableRequest =
+          CreateTableRequest.of(TABLE_ID).addFamily(COLUMN_FAMILY_NAME);
+      adminClient.createTable(createTableRequest);
+
+      try (BigtableDataClient dataClient = BigtableDataClient.create(projectId, instanceId)) {
```

Comment:
Should there be a discussion of key design?  Having a common prefix is ok for tests, but it's exactly the pattern we are trying to discourage users from doing.

Reply:
The schema is deviceType#deviceId#YYYYMMDD this was discussed and reviewed with members of the Bigtable team. This is only a small subset of data so we can't really show much variety